arrow/util.py

```diff
@@ -23,6 +28,44 @@ def is_timestamp(value):
         return False
 
 
+def windows_datetime_from_timestamp(timestamp, tz=None):
+    """Computes datetime from timestamp. Supports negative timestamps on Windows platform."""
+    sec_frac, sec = math.modf(timestamp)
+    dt = datetime.datetime(1970, 1, 1, tzinfo=dateutil.tz.tzutc()) + datetime.timedelta(
+        seconds=sec, microseconds=sec_frac * 1000000
+    )
+    if tz is None:
+        tz = dateutil.tz.tzlocal()
+
+    if tz == dateutil.tz.tzlocal():
```

Comment:
Thanks for making the requested changes @kkoziara! This is a great PR and it works well from what I have tried locally!

Before we merge this in, think you can explain the content of this conditional a bit more (lines 40-47)?

Reply:
Lines 40-47 adjust the `dt` datetime computed using `timedelta` for the local timezone. Normally it's possible to use `datetime.astimezone` method for this, but on Windows for `tz.tzlocal()` and dates before the epoch this does not work. That's why I made this convoluted workaround. Here:

        ts = time.mktime(dt.replace(year=1970).timetuple())
 
I get the non-negative timestamp with the same day of the year as `dt` to have the same daylight saving time as `dt`. Then I use:

        dt += datetime.datetime.fromtimestamp(ts) - datetime.datetime.utcfromtimestamp(
            ts
        )

to get the offset to the UTC for the local timezone and proper daylight saving time, as original `fromtimestamp` will work here (`ts` is non-negative). Then I modify the computed `dt` with the offset. Finally in here: `dt = dt.replace(tzinfo=dateutil.tz.tzlocal())`  we just set the proper `tzinfo`.

I know this is not very straightforward but I didn't get any simpler ideas to achieve this. I am open to change this to something simpler.

Reply:
I think it is fine--it is a fantastic workaround.